internal/service/athena/data_catalog.go

```diff
@@ -0,0 +1,254 @@
+package athena
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"regexp"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/arn"
+	"github.com/aws/aws-sdk-go/service/athena"
+	"github.com/hashicorp/aws-sdk-go-base/tfawserr"
+	"github.com/hashicorp/go-cty/cty"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+	"github.com/hashicorp/terraform-provider-aws/internal/flex"
+	tftags "github.com/hashicorp/terraform-provider-aws/internal/tags"
+	"github.com/hashicorp/terraform-provider-aws/internal/verify"
+)
+
+func ResourceDataCatalog() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceDataCatalogCreate,
+		ReadContext:   resourceDataCatalogRead,
+		UpdateContext: resourceDataCatalogUpdate,
+		DeleteContext: resourceDataCatalogDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		CustomizeDiff: verify.SetTagsDiff,
+
+		Schema: map[string]*schema.Schema{
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.All(
+					validation.StringLenBetween(1, 129),
+					validation.StringMatch(regexp.MustCompile(`[\w@-]*`), ""),
+				),
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
```

Comment:
The API documentation states that `description` is not required, yet it does fail the validation it the `description` is not set with a minimum length of 1.

I won't implement any validation on our side as I want to align with the API documentation. When applying without a `description` the user gets a error message that the minimum lenghth should be 1.